src/Compilers/Core/Portable/InternalUtilities/ReadOnlyHashSet.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Roslyn.Utilities
+{
+    internal struct ReadOnlyHashSet<T> : IReadOnlySet<T>
```

Comment:
Then can you move this into SpecializedCollections and change that return type as well?

---
In reply to: [74471084](https://github.com/dotnet/roslyn/pull/12353#discussion_r74471084) [](ancestors = 74471084)